client/pool-buffer.c

```diff
@@ -103,30 +103,29 @@ void destroy_buffer(struct pool_buffer *buffer) {
 }
 
 struct pool_buffer *get_next_buffer(struct wl_shm *shm,
-		struct pool_buffer pool[static 2], uint32_t width, uint32_t height) {
-	struct pool_buffer *buffer = NULL;
-
-	for (size_t i = 0; i < 2; ++i) {
-		if (pool[i].busy) {
-			continue;
-		}
-		buffer = &pool[i];
-	}
-
-	if (!buffer) {
-		return NULL;
```

Comment:
This return should already handle when both buffers are busy, as buffer then remains NULL.

The only change I can see is when both buffers are available, and the second buffer matches the size while the first does not. Is this the scenario you intended to optimize?

Reply:
Thank you for your review! The code snippet isn't specifically tailored to handle scenarios where both buffers are busy. It primarily addresses the condition where buffer remains NULL if unassigned. The optimization aims to improve buffer allocation efficiency, especially focusing on scenarios where both buffers are available but only one matches the required size. If that makes sense.

Reply:
I misread your message originally. Yes that is the intended scenario
.